### PR TITLE
fix ALLUXIO-2036

### PR DESCRIPTION
**Pull request by lordk911**
I have tested with : mvn clean package -Phadoop-2.7 -Pspark -rf :alluxio-core-client

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, ok to test

**Comment by aaudiber** [comment]
Hi @lordk911, thanks for adding this test. Could you update the PR message to summarize the change you made?

**Comment by aaudiber** [comment]
Also, please include a link to the JIRA ticket in the PR message

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 1]
```diff
@@ -159,6 +159,33 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
     alluxio.client.file.FileSystem.Factory.get();
     ClientTestUtils.resetClientContext();
   }
+  
+  /**
+  * Tests for ALLUXIO-2036.
+  * Cause by ALLUXIO-2025
+  */
+  @Test
+  public void shouldThrowFileNotFoundExceptionWhenListStatus4NoneExistFile() throws Exception {
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: could we shorten this to `ListStatusNonExisting`?

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 2]
```diff
@@ -159,6 +159,33 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
     alluxio.client.file.FileSystem.Factory.get();
     ClientTestUtils.resetClientContext();
   }
+  
+  /**
+  * Tests for ALLUXIO-2036.
+  * Cause by ALLUXIO-2025
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: The javadoc should describe what is being tested, not the ticket for writing the test

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 3]
```diff
@@ -159,6 +159,33 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
...
+  */
+  @Test
+  public void shouldThrowFileNotFoundExceptionWhenListStatus4NoneExistFile() throws Exception {
+    // final Configuration conf = new Configuration();
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: remove the dead line

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 4]
```diff
@@ -159,6 +159,33 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
...
+    StringBuilder path = new StringBuilder("/ALLUXIO-2036.");
+    path.append(System.currentTimeMillis()).append(".txt");
+    FileSystem fs = new FileSystem();
+    try {
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: The normal pattern we use to check that an exception is thrown looks like
```java
try {
  fs.listStatus(new Path(path.toString()));
  Assert.fail("Listing the status of a nonexistent file should throw an exception");
} catch (FileNotFoundException e) {
  // optionally check that e.getMessage() is correct
} finally {
  fs.close();
}
```

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 5]
```diff
@@ -159,6 +159,33 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
...
+      Assert.assertTrue(e instanceof FileNotFoundException);
+    } finally {
+      try {
+        fs.close();
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    }
+
+    PowerMockito.verifyStatic();
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: I don't think these last three lines should be necessary

**Comment by aaudiber** [comment]
It looks like the build is failing because of trailing whitespace on line 162

**Comment by lordk911** [comment]
thank you . and I am sorry that it seems like I make trouble more than I want to resolve.

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 6]
```diff
@@ -161,6 +161,24 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
   }
 
   /**
+  * Ensures that when ListStatusNonExisting should Throw FileNotFoundException.
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: `Ensures that FileNotFoundException is thrown when listStatus is called on a nonexistent path.`

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 7]
```diff
@@ -161,6 +161,24 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
...
+  */
+  @Test
+  public void shouldThrowFileNotFoundExceptionWhenListStatusNonExisting() throws Exception {
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: We typically end our test names with `Test`. The "should" is implicit in all tests, so we don't need to include it as part of the test name.

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 8]
```diff
@@ -161,6 +161,24 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
...
+    StringBuilder path = new StringBuilder("/ALLUXIO-2036.");
+    path.append(System.currentTimeMillis()).append(".txt");
+    FileSystem fs = new FileSystem();
+    try {
+      fs.listStatus(new Path(path.toString()));
+      Assert.fail("Listing the status of a nonexistent file should throw an exception");
+    } catch (FileNotFoundException e) {
+      LOG.info(e.getMessage());
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: This can just be `// This exception is expected` instead of logging. Logging could make it look like there was an error even when the test passes.

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 9]
```diff
@@ -161,6 +161,24 @@ public void hadoopShouldLoadFileSystemWhenConfiguredTest() throws Exception {
   }
 
   /**
+  * Ensures that when ListStatusNonExisting should Throw FileNotFoundException.
+  */
```
Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: I think this line and the one above it should be indented one further

**Comment by aaudiber** [comment]
@lordk911 I left a couple more minor comments, thanks for the quick round of iteration. The build seems to be failing due to missing imports:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java:[172,25] cannot find symbol
  symbol:   class Path
  location: class alluxio.hadoop.AbstractFileSystemTest
[ERROR] /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java:[174,14] cannot find symbol
  symbol:   class FileNotFoundException
  location: class alluxio.hadoop.AbstractFileSystemTest
```

**Comment by lordk911** [comment]
I don't know why listStatus(AlluxioURI uri) method didn't throw out any exception.

**Comment by aaudiber** [comment]
@lordk911 do you mind investigating why the test doesn't work?

**Comment by lordk911** [comment]
@aaudiber OK,I have change the code and can run unit test passed.

**Comment by aaudiber** [comment]
Thanks for fixing, LGTM
